File: 프로그래머스/lv2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    let answer = 0;

    people.sort((a, b) => b - a); // 내림차순으로 정렬

    // i는 왼쪽 인덱스, j는 오른쪽 인덱스
    for (let i = 0, j = people.length - 1; i <= j; i++) {
        // 왼쪽 사람과 오른쪽 사람의 무게 합이 limit 이하인 경우
        if (people[i] + people[j] <= limit)
            j--; // 오른쪽 인덱스를 한 칸 앞으로 이동하여 다음으로 무거운 사람과 짝을 이룸
        answer++; // 한 쌍의 사람을 구출했으므로 구출된 사람 수 증가
    }

    return answer; // 구출된 사람 수 반환
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    answer = 0
    # 처음 참 고민했는데 2명씩이란 말을 늦게 봄
    # 무게 낮은 값부터 정렬
    people.sort()
    l = len(people)
    start = 0
    end = l-1
    # 모두 보트를 탈때까지
    while start <= end:
        if people[start] + people[end] <= limit:
            start += 1
            end -= 1
        else:
            end -= 1
        answer += 1
        
    return answer